src/kiara_plugin/topic_modelling/modules/corpus_preparation.py

```diff
@@ -109,4 +110,71 @@ def get_date(file):
         except Exception as e:
             print(f"An error occurred: {e}")
 
-        outputs.set_value("corpus_table", output_table)
\ No newline at end of file
+        outputs.set_value("corpus_table", output_table)
+
+class CorpusDistTime(KiaraModule):
+    """
+    This module aggregates a table by day, month or year from a corpus table that contains a date column. It returns the distribution over time, which can be used for display purposes, such as visualization.
+
+    Dependencies:
+    - polars: https://www.pola.rs/
+    - pyarrow: https://arrow.apache.org/docs/python/
+    - duckdb: https://duckdb.org/
+    
+    """
+
+    _module_type_name = "topic_modelling.time_dist"
+
+    def create_inputs_schema(self):
+        
+        return {
+             "periodicity": {
+                "type": "string",
+                "doc": "The desired data periodicity to aggregate the data. Values can be either 'day','month' or 'year'."
+            },
+            "date_col_name": {
+                "type": "string",
+                "doc": "The column that contains the date, a list of compliant formats can be consulted here: https://docs.rs/chrono/latest/chrono/format/strftime/index.html."
+            },
+            "title_col_name": {
+                "type": "string",
+                "doc": "The column that contains publication names or ref/id."
+            },
+            "corpus_table": {
+                "type": "table",
+                "doc": "The corpus table for which the distribution over time is needed."
+            }
+        }
+
+    def create_outputs_schema(self):
+        return {
+            "dist_table": {
+                "type": "table",
+                "doc": "The aggregated data table."
+            }
+        }
+
+    def process(self, inputs, outputs) -> None:
+
+        agg = inputs.get_value_obj("periodicity").data
+        title_col = inputs.get_value_obj("title_col_name").data
+        time_col = inputs.get_value_obj("date_col_name").data
+        
+        table_obj = inputs.get_value_obj("corpus_table")
+        
+        sources = pl.from_arrow(table_obj.data.arrow_table)
+
+        sources = sources.with_columns(pl.col(time_col).str.strptime(pl.Date, "%Y-%m-%d"))
```

Comment:
How does the exception look like if the `time_col` doesn't exist in the table? I'd assume fairly cryptic? It'd be better to test if it exists first, and if not raise a `KiaraProcessingException` that tells the user the column name is invalid, and ideally which column names are available as well.

Also, the `"%Y-%m-%d"` is a fixed, expected format? What happens if the data contains the date in a different format? We'd have to assume there are all sorts of date formats in the data users come to us with, so it probably either needs some sort of user input, and optionally we could implement some sort of auto-detect if a user does not provide the format. The latter one could still fail, so it would need a good exception message.

Reply:
Also, I don't quite understand why you use polars here to parse the date, and below duckdb for the query. Wouldn't it be better to integrate the date parsing directly into the query using sometihng like https://duckdb.org/docs/sql/functions/dateformat.html ? 

That way, the data doesn't need to be touched twice.

Reply:
> How does the exception look like if the time_col doesn't exist in the table? I'd assume fairly cryptic? It'd be better to test if it exists first, and if not raise a KiaraProcessingException that tells the user the column name is invalid, and ideally which column names are available as well.

Indeed, where can I consult what can be done with KiaraProcessingException, is the code for this located in the Kiara repo or is it in a Kiara plugin?

Reply:
> Also, I don't quite understand why you use polars here to parse the date, and below duckdb for the query. Wouldn't it be better to integrate the date parsing directly into the query using sometihng like https://duckdb.org/docs/sql/functions/dateformat.html ?
> 
> That way, the data doesn't need to be touched twice.

The problem is that I do not know in advance which format users have for their date. My understanding was that Polars `pl.date` can handle many formats https://stuffbyyuki.com/how-to-convert-string-to-date-or-datetime-in-polars/ if they comply with this: https://docs.rs/chrono/latest/chrono/format/strftime/index.html. It is then possible to specify the conversion target format, like e.g. here `%Y-%m-%d`, without having to specifying the input date format:

`sources = sources.with_columns(pl.col(time_col).str.strptime(pl.Date, "%Y-%m-%d"))`

I may have understood this wrong, but it seems that in duckdb the input format needs to be specified, no? https://duckdb.org/docs/sql/functions/dateformat.html. That said, I will do some tests to confirm these assumptions (but maybe you already tested yourself already?) and see if maybe duckdb does the same thing and can handle dates without specifying the input format.

Reply:
Actually, forget that, just raise any suitable exception, kiara will wrap it into a `KiaraProcessingException` anyway.

Reply:
I resolve this following your very last comment on this.